File: 2011/python/2011.py
# python 2011.py

import re

class Solution(object):
    def finalValueAfterOperations(self, operations):
        """
        :type operations: List[str]
        :rtype: int
        """
        xVal = 0

        for e in operations:
            xVal += 1 if re.match("^.*\++.*$", e) else -1

        return xVal


solution = Solution()

op1 = ["++X", "++X", "X++"]
op2 = ["--X", "X++", "X++"]

print(solution.finalValueAfterOperations(op1))
print(solution.finalValueAfterOperations(op2))

File: 2011/javascript/2011.js
// tsc 2011.js

function finalValueAfterOperations(operations) {
    let xVal = 0;

    operations.map((str) => {
        xVal += str.match(/^.*\++.*$/) ? 1 : -1;
    });

    return xVal;
};

console.log(finalValueAfterOperations(["++X", "++X", "X++"]));
console.log(finalValueAfterOperations(["--X", "X++", "X++"]));
